src/renderer/render_info.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from "blockly/core";
+import { BowlerHat } from "./bowler_hat.js";
+
+export class RenderInfo extends Blockly.zelos.RenderInfo {
+  populateTopRow_() {
+    if (this.isBowlerHatBlock()) {
+      const bowlerHat = new BowlerHat(this.constants_);
+      this.topRow.elements.push(
+        new Blockly.blockRendering.SquareCorner(this.constants_)
+      );
+      this.topRow.elements.push(bowlerHat);
+      this.topRow.elements.push(
+        new Blockly.blockRendering.SquareCorner(this.constants_)
+      );
+      this.topRow.minHeight = 0;
+      this.topRow.capline = bowlerHat.ascenderHeight;
+    } else {
+      super.populateTopRow_();
+    }
+  }
+
+  populateBottomRow_() {
+    super.populateBottomRow_();
+    if (this.isBowlerHatBlock()) {
+      this.bottomRow.minHeight = this.constants_.MEDIUM_PADDING;
+    }
+  }
+
+  computeBounds_() {
+    super.computeBounds_();
+    if (this.isBowlerHatBlock()) {
+      // Resize the render info to the same width as the widest part of a
+      // bowler hat block.
+      const statementRow = this.rows.find((r) => r.hasStatement);
+      this.width =
+        statementRow.widthWithConnectedBlocks -
+        statementRow.elements.find((e) =>
+          Blockly.blockRendering.Types.isInput(e)
+        ).width +
```

Comment:
Why are you subtracting the width of the inner statement block's input? `statementRow` is the block that lets you see what your caller will look like right?

Reply:
Herm it feels very likely to that this is not the thing we want to be subtracting, and instead the width of the "define" text just happens to match some other constant somewhere that makes this look correct. If you change the "define" text to "supercalifragilisticexpialidocious" does the math still work correctly?

Reply:
Yep!
<img width="474" alt="Screenshot 2024-08-08 at 2 30 40 PM" src="https://github.com/user-attachments/assets/8346a639-2b74-4d4a-8b90-9cc336955b56">

Reply:
sweetness sgtm then!